Fix building gspeak on macOS

mohd-akram opened the pull request:
Fix some warnings that are errors on modern Clang, specifically `incompatible-function-pointer-types`, `implicit-int` and `implicit-function-declaration`. Also, ensure the build uses `libtts.dylib` on macOS and fix paths to resources.

stargo commented on src/samplosf/src/windict/compile_dict.c:
```diff
@@ -263,6 +263,15 @@ void pushEntry();
+int parseString(char*);
+void scan_entry();
+void sort_entry();
+void scan_error(char*);
```
This is correct, but causes an error in line 844 later on as scan_error is called there with a vararg argument. Can you fix line 844 to only call with one arg (maybe sprintf to a temporary buffer or so)?

mohd-akram replied:
It seemed to build with that, so I just left it :D I've added a sprintf there.

mohd-akram committed:
Fix building gspeak on macOS

Fix some warnings that are errors on modern Clang, specifically
`incompatible-function-pointer-types`, `implicit-int` and
`implicit-function-declaration`. Also, ensure the build uses
`libtts.dylib` on macOS and fix paths to resources.

stargo commented:
Thanks, merging :-)
